syncer/main.py

```diff
@@ -54,137 +127,392 @@ def __init__(self, webinterface, connector):
+                        shutters_to_rename.append(shutter)
+            else:
+                shutters_to_rename = [remote_shutter_conf[remote_id]]
+
+            for shutter in shutters_to_rename:
+                old_name = shutter.get('name')
+                if ' (also controlled by syncer plugin ' in old_name:
+                    old_name = old_name.split(' (also controlled by syncer plugin ')[0]
+                remote_shutter_conf[shutter.get("id")][
+                    'name'] = f"{old_name} (also controlled by syncer plugin {self._name}/{local_shutter_name})"
+                self._call_remote(
+                    f"set_shutter_configuration?config={json.dumps(remote_shutter_conf[shutter.get('id')])}",
+                    gateway=self._gateways[ip])
+        except Exception as ex:
+            raise RuntimeError(
+                f'Could not set configuration remote shutter{"group" if is_shutter_group else ""} {ip}/{remote_id}: {ex}')
+
+    def update_remote_shutter_state(self, ip, remote_id, state, reverse, is_shutter_group):
+        try:
+            do_call = True
+            if state == "stopped":
+                remote_state = "stop"
+            elif state == "going_up":
+                remote_state = "up" if not reverse else "down"
+            elif state == "going_down":
+                remote_state = "up" if reverse else "down"
+            else:
+                remote_state = ""
+                do_call = False
+
+            if is_shutter_group:
+                group = "group_"
+            else:
+                group = ""
+
+            if do_call:
+                self._call_remote(api_call=f"do_shutter_{group}{remote_state}", params={
+                    "id": remote_id
+                }, gateway=self._gateways[ip])
+        except Exception as ex:
+            raise RuntimeError(
+                f'Could not set state of remote shutter{"group" if is_shutter_group else ""} {ip}/{remote_id}: {ex}')
 
     @background_task
     def run(self):
-        previous_values = {}
         while True:
             if not self._enabled:
                 time.sleep(30)
                 continue
             try:
                 # Sync sensor values:
-                data_humidities = json.loads(self.webinterface.get_sensor_humidity_status())
-                data_temperatures = json.loads(self.webinterface.get_sensor_temperature_status())
-                if data_humidities['success'] is True and data_temperatures['success'] is True:
-                    for sensor_id in range(len(data_temperatures['status'])):
-                        if sensor_id not in self._sensor_mapping:
+                for gateway in self._gateways.values():
+                    sensor_mapping = gateway.get('sensor_mapping')
+                    if sensor_mapping == {}:
+                        continue
+                    sensor_status = self._call_remote('get_sensor_status', gateway).get('status')
```

Comment:
Something to tweak in the future: This is a polling base system. It would be better to push events to the known gateways instead of polling the sensors every interval.

Reply:
This would mean local to remote, which is currently not possible as we can't make a virtual sensor object on the remote GW. This is only possible with connectors and we can't reach them from remote, API-calls don't support this. Unless you know another way?